internal/backend/remote-state/azure/backend.go

```diff
@@ -49,6 +49,13 @@ func New() backend.Backend {
 				Description: "The blob key to use in the Storage Container.",
 			},
 
+			"lookup_blob_endpoint": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ARM_USE_DNS_ZONE_ENDPOINT", false),
+				Description: "Whether to look up the storage account blob endpoint, instead of composing the endpoint in a fixed pattern. This is necessary when the storage account uses the Azure DNS zone endpoint.",
```

Comment:
I'm unclear on what this line means `instead of composing the endpoint in a fixed pattern`. I'd recommend expanding on what that means more or just removing it
```suggestion
				Description: "Whether to look up the storage account blob endpoint. This is necessary when the storage account uses the Azure DNS zone endpoint.",
```

Reply:
Removed. The endpoint is always needed, if not looking it up, we'll construct it using the fixed pattern: ``https://<storage-account>.blob.core.windows.net`, which ends up with URL `https://<storage-account>.blob.core.windows.net/<containerName>/<key>` for a blob.